Gems/EMotionFX/Code/EMotionFX/Rendering/OpenGL2/Source/GLActor.cpp

```diff
@@ -64,42 +55,36 @@ namespace RenderGL
     // get rid of the allocated memory
     void GLActor::Cleanup()
     {
-        uint32 i;
-
         // get rid of all index and vertex buffers
-        for (uint32 a = 0; a < 3; ++a)
+        for (AZStd::vector<VertexBuffer*>& vertexBuffers : mVertexBuffers)
         {
-            // get rid of the given vertex buffers
-            const uint32 numVertexBuffers = mVertexBuffers[a].GetLength();
-            for (i = 0; i < numVertexBuffers; ++i)
+            for (VertexBuffer* vertexBuffer : vertexBuffers)
             {
-                delete mVertexBuffers[a][i];
+                delete vertexBuffer;
             }
-
-            // get rid of the given index buffers
-            const uint32 numIndexBuffers = mIndexBuffers[a].GetLength();
-            for (i = 0; i < numIndexBuffers; ++i)
+        }
+        for (AZStd::vector<IndexBuffer*>& indexBuffers : mIndexBuffers)
+        {
+            for (IndexBuffer* indexBuffer : indexBuffers)
             {
-                delete mIndexBuffers[a][i];
+                delete indexBuffer;
             }
         }
 
         // delete all materials
-        const uint32 numLOD = mMaterials.GetLength();
-        for (uint32 l = 0; l < numLOD; l++)
+        for (AZStd::vector<MaterialPrimitives*>& materialsPerLod : mMaterials)
         {
-            const uint32 numMaterials = mMaterials[l].GetLength();
-            for (uint32 n = 0; n < numMaterials; n++)
+            for (MaterialPrimitives* materialPrimitives : materialsPerLod)
             {
-                delete mMaterials[l][n]->mMaterial;
-                delete mMaterials[l][n];
+                delete materialPrimitives->mMaterial;
+                delete materialPrimitives;
             }
         }
     }
```

Comment:
The function is actually not clearing `mVertexBuffers`, `mIndexBuffers` and `mMaterials`, leaving the elements with deleted pointers. I guess since it was working before it's not necessary?

Reply:
Yeah, you're right. `Cleanup()` is called from the destructor, so in that case it doesn't matter if the vector is cleared or not, as the vector's destructor itself will do that. `Cleanup()` is also called in case the initialization of the object failed, in which case (probably) none of its methods are ever called. So since it has been working this way for a while, I don't see the need to change it as part of this.